cachi2/core/package_managers/generic/main.py

```diff
@@ -47,8 +54,42 @@ def _resolve_generic_lockfile(source_dir: RootedPath, output_dir: RootedPath) ->
 
     log.info(f"Reading generic lockfile: {lockfile_path}")
     lockfile = _load_lockfile(lockfile_path)
+    to_download: dict[str, Union[str, os.PathLike[str]]] = {}
+
+    # output_dir is now the root and cannot be escaped
+    output_dir = output_dir.re_root(DEFAULT_DEPS_DIR)
+
     for artifact in lockfile.artifacts:
         log.debug(f"Resolving artifact: {artifact.download_url}")
+        # make sure target does not point outside the output directory
```

Comment:
nitpick: I think this commentary is redundant as we pretty much use `join_within_root` for this particular purpose across the whole code base.

Reply:
Removed